langchain/llms/gpt4all.py

```diff
@@ -1,7 +1,9 @@
 """Wrapper for the GPT4All model."""
+import io
 from functools import partial
 from typing import Any, Dict, List, Mapping, Optional, Set
 
+import pkg_resources
```

Comment:
I think we should opt for a different stdlib package if possible https://discuss.python.org/t/will-setuptools-remove-pkg-resource-module-in-the-future/27182

Reply:
@hinthornw We can do it in another way, the streaming for gpt4all throws `NotImplementedError` if used the previous version, hence we can detect there by a `try-except` block. In that way we can do it without `pkg_resource` 

Another thing, should we remove the callback support for streaming and make this changes as the default one? In that way there might be less ambiguity?

Reply:
@hinthornw I also wonder, Does providing custom StreamingCallback provide any kind of added advantage?